host/tests/L3_TestCases/dsVideoDevice/dsVideoDevice_test4_SetAndGetFRFMode.py

```diff
@@ -0,0 +1,142 @@
+
+        Args:
+            None.
+        """
+        self.testName  = "test4_FRFMode"
+        super().__init__(self.testName, '1')
+
+
+    def testVerifyFRFMode(self, manual=False, frfmode=False):
+        """
+        Verifies the FRF mode on specified video device.
+
+
+        Args:
+            manual (bool, optional): If True, requires manual confirmation from the user.
+                                     Defaults to False.
+
+        Returns:
+            bool: True if selected FRF mode is visible in output device; otherwise, False.
+        """
+        if manual == True and frfmode == True:
+            return self.testUserResponse.getUserYN(f"Wait till playback ends. And, Check whether the video playback is smooth without any streaming issues (Y/N):")
+        elif manual == True and frfmode == False:
+            return self.testUserResponse.getUserYN(f"Wait till playback ends. And, Check whether the video playback has streamming issues is seen? (Y/N):")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    
+    def testFunction(self):
+        """
+        Executes the set and get FRF mode tests on the video device.
+
+        This function performs the following steps:
+        - Downloads necessary assets.
+        - Runs prerequisite commands.
+        - Initializes the dsVideoDevice module.
+        - Get the number of video devices
+        - Set the FRF mode to Enable
+        - Play the video stream
+        - Wait till playback ends and verify with observation
+        - Set the FRF mode to Disable
+        - Play the video stream
+        - Wait till playback ends and verify with observation
+        - Cleans up the downloaded assets after testing.
+
+        Returns:
+            bool: Status of the last verification (True if successful, False otherwise).
+        """
+        
+        self.log.testStart(self.testName, '1')
+
+        # Initialize the dsVideoDevice module
+        self.testdsVideoDevice.initialise(self.testdsVideoDevice.getDeviceType())
+
+        NumofVideoDevices = self.testdsVideoDevice.getVideoDevice()
+        for i in range(0, NumofVideoDevices):
+            self.testdsVideoDevice.setFRFMode(i, 'Enable')
+            self.log.step(f'Wait till Playback ends.')
+            
+            #Playback with framerate 24fps
+            for stream in self.testStreams:
+                self.testPlayer.play(stream)
+
+            result = self.testVerifyFRFMode(True,True)
+            if result:
+                self.log.stepResult(result, f'Playback was fine without any streaming issues')
+            else:
+                self.log.error("Able to observe streaming issues")
+
+            self.testdsVideoDevice.setFRFMode(i, 'Disable')
+            self.log.step(f'Wait till Playback ends.')
+            
+            #Playback with framerate 24fps
+            for stream in self.testStreams:
+                self.testPlayer.play(stream)
```

Comment:
Here the player plays the different streams in loop without any checks

Reply:
Updated